cert/aziot-certd/src/lib.rs

```diff
@@ -394,7 +397,8 @@ fn create_cert<'a>(
                         ))
                     })?;
 
-                    let url = urls
+                    let auth = cert_auth.as_ref().unwrap_or(&default_auth);
+                    let url = cert_url.as_ref().unwrap_or(urls
```

Comment:
`.unwrap_or_else(|| urls.get(id)).or_else(|| ...` so that it's evaluated lazily.